panels/site_sections/panel_applications.py

```diff
@@ -32,34 +55,72 @@ def index(self, session, message='', **params):
         panelist = session.panel_applicant(panelist_params, checkgroups=PanelApplicant.all_checkgroups, restricted=True, ignore_csrf=True)
         panelist.application = app
         panelist.submitter = True
-        other_panelists = []
-        for i in range(1, int(params.get('other_panelists', 0)) + 1):
-            applicant = {attr: params.get('{}_{}'.format(attr, i)) for attr in OTHER_PANELISTS_FIELDS}
-            other_panelists.append(PanelApplicant(application=app, **applicant))
 
         if cherrypy.request.method == 'POST':
-            message = check(panelist) or check(app) or check_other_panelists(other_panelists)
+            message = check(panelist) or check_extra_verifications(int(params.get('other_panelists', 0)), **params)
             if not message:
-                if 'verify_unavailable' not in params:
-                    message = 'You must check the box to confirm that you are only unavailable at the specified times'
-                elif 'verify_waiting' not in params:
-                    message = 'You must check the box to verify you understand that you will not hear back until {}'.format(c.EXPECTED_RESPONSE)
-                elif 'verify_tos' not in params:
-                    message = 'You must accept our Terms of Accomodation'
-                elif other_panelists and 'verify_poc' not in params:
-                    message = 'You must agree to being the point of contact for your group'
-                else:
-                    session.add_all([app, panelist] + other_panelists)
-                    raise HTTPRedirect('index?message={}', 'Your panel application has been submitted')
+                message = process_panel_app(session, app, panelist, **params)
+                if not message:
+                    raise HTTPRedirect('guest?poc_id={}message={}', poc_id, 'Your panel application has been submitted')
 
         return {
             'app': app,
             'message': message,
             'panelist': panelist,
-            'ops_count': len(other_panelists),
```

Comment:
The `ops_count` is just replaced by a `other_panelists|length` since it's used only once in the template.